bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/composite/CompositeModuleHandlerFactory.java

```diff
@@ -78,87 +78,102 @@ public CompositeModuleHandlerFactory(BundleContext bc, ModuleTypeManager mtManag
 
     @SuppressWarnings({ "unchecked" })
     @Override
-    public void ungetHandler(Module module, String ruleUID, ModuleHandler handler) {
-        ModuleHandler handlerOfModule = handlers.get(ruleUID + module.getId());
+    public void ungetHandler(Module module, String childModulePrefix, ModuleHandler handler) {
+        ModuleHandler handlerOfModule = handlers.get(childModulePrefix + module.getId());
```

Comment:
In all other places (other handlers) there is not a colon (Please see handlers map of BaseModuleHandlerFactory. I don't know why, may be to have less string concatenations). In all these places  ruleId + moduleId is unique  identifier of the handler. 
The RuleEngine has a mapping between module type and list of rules where the module type is  used. (In that way it changes rule state not_initialized <-> initialized) 
A rule which have composite module type depends on its all child module types. There for the composite module type has to update Rule Engine's module type to rule mapping. 
The colon is used to separate rule id from the rest of prefix, because if you have a nested composite module types (i.e. ItemStateChangeTrigger) the method is called in recursion and rule id will be loosed. For that reason the composite module type uses colon in definition of unique id of handler.